### DISCUSSION TITLE
Retrieving search results from engines

### DISCUSSION THREAD

**Discussion by theonlyalm** [searxng/searxng#3371] [Q&A]
What are the techniques used to retrieve search results from engines? Is it similar to web scraping?

And what are the files responsible for retrieving data from engines? Cuz actually I'm confused with all these files

**Comment by Austin-Olacsi** [comment]
The files responsible for retrieving search results are the .py files in the searx/engines folder.

Sometimes web scraping is used, sometimes APIs are used, it just depends on the engine.

**Reply by theonlyalm** [reply]
So what about using C++ in scraping, wouldn't it be better?

And what's the file responsible for combining all these search results together and prioritizing them?
